Source/WebCore/platform/graphics/cocoa/GraphicsContextGLCocoa.mm

```diff
@@ -578,15 +579,33 @@ static bool needsEAGLOnMac()
     return allocateAndBindDisplayBufferBacking();
 }
 
+DestinationColorSpace GraphicsContextGLCocoa::getDrawingBufferColorSpace()
+{
+    return m_drawingBufferColorSpace;
+}
+
+void GraphicsContextGLCocoa::setDrawingBufferColorSpace(const DestinationColorSpace& colorSpace)
+{
+    if (m_drawingBufferColorSpace != colorSpace) {
+        m_drawingBufferColorSpace = colorSpace;
+
+        if (!getInternalFramebufferSize().isEmpty() && !reshapeDisplayBufferBacking()) {
+            RELEASE_LOG(WebGL, "Fatal: Unable to allocate backing store of size %d x %d", getInternalFramebufferSize().width(), getInternalFramebufferSize().height());
+            forceContextLost();
+        }
+    }
+}
+
 bool GraphicsContextGLCocoa::allocateAndBindDisplayBufferBacking()
 {
     ASSERT(!getInternalFramebufferSize().isEmpty());
-    auto backing = IOSurface::create(nullptr, getInternalFramebufferSize(), DestinationColorSpace::SRGB());
+    auto backing = IOSurface::create(nullptr, getInternalFramebufferSize(), m_drawingBufferColorSpace);
     if (!backing)
         return false;
     if (m_resourceOwner)
         backing->setOwnershipIdentity(m_resourceOwner);
     backing->migrateColorSpaceToProperties();
+    m_drawingBufferColorSpace = backing->colorSpace();
```

Comment:
I think here you have a redundant assignment. If this was an attempt to address the comments above, the issue is the inverse of this, not this.

Reply:
My idea was that we first assign `m_drawingBufferColorSpace` with the requested value, but then the "backing" IOSurface could choose another colorspace based on its capabilities, and we would record that actual value.
And that's why there are separate set&get functions, because some platforms could refuse a set("display-p3") and that would be discoverable by doing a get() that would return "srgb" -- does that make sense?

> If this was an attempt to address the comments above, the issue is the inverse of this, not this.

Sorry, I'm getting a bit lost in all these comments, which comment/issue/invers are you referring to?

My goal was to get close to how the JS interface (i.e., the `drawingBufferColorSpace` attribute) works: Set a value, but it may not actually be accepted in the engine; reading back `drawingBufferColorSpace` reveals what is really set.
But now that makes me think that if we try to set an invalid colorspace, the IOSurface would not be created, but then we should restore what was there before. It's getting harder by the minute!
If it's too awkward to design something through these comments, maybe we should chat first.